sqldelight-compiler/environment/build.gradle

```diff
@@ -0,0 +1,122 @@
+plugins {
+  alias(libs.plugins.kotlin.jvm)
+  alias(libs.plugins.shadow)
+  alias(libs.plugins.publish)
+  id("app.cash.sqldelight.toolchain.compiler")
+}
+
+dependencies {
+  shadow(libs.intellij.core) {
+    transitive = true
+  }
+  shadow(libs.intellij.analysis) {
+    // Only shade the minimum IntelliJ libs.
+    transitive = false
+  }
+  shadow(libs.intellij.lang) {
+    transitive = false
+  }
+  shadow(libs.intellij.coreImpl) {
+    transitive = false
+  }
+  shadow(libs.intellij.coreUi) {
+    transitive = false
+  }
+  shadow(libs.intellij.projectModel) {
+    transitive = false
+  }
+  shadow(libs.intellij.projectModelImpl) {
+    transitive = false
+  }
+  shadow(libs.intellij.analysisImpl) {
+    transitive = false
+  }
+  shadow(libs.intellij.testFramework) {
+    transitive = false
+  }
+  shadow(libs.intellij.langImpl) {
+    transitive = false
+  }
+  shadow(libs.intellij.ide) {
+    transitive = false
+  }
+  shadow(libs.intellij.ideCore) {
+    transitive = false
+  }
+  shadow(libs.intellij.ideImpl) {
+    transitive = false
+  }
+  shadow(libs.intellij.util) {
+    transitive = false
+  }
+  shadow(libs.intellij.utilUi) {
+    transitive = false
+  }
+  shadow(libs.intellij.utilEx) {
+    transitive = false
+  }
+
+  shadow(libs.jna)
+}
+
+tasks.named("jar") {
+  // shadowjar overwrites the normal jar, this overlapping outputs are not cacheable.
+  // and we don't need the non shaded jar.
+  enabled = false
+}
+
+tasks.named("shadowJar") {
+  archiveClassifier.set("")
+  configurations = [project.configurations.shadow]
+
+  minimize {
```

Comment:
Todo: tune this list to reduce the shaded artifact, currently 43 MB.

Reply:
With shadow, further size reduction is not possible, because shadow uses artifacts to minimize jars. The biggest jars are `lang-impl` (19 MB) and `ide-impl` (18 MB). The resulting jar is still 43 MB, but without minimizing it is 67 MB.